stuff

In [1]:
import pandas as pd
import numpy as np
import re 

%pprint            # to turn off pretty printing

Pretty printing has been turned OFF


In [2]:
# I tried to use .read_csv, but it didn't work, so I'll continue to use .from_csv
balc = pd.DataFrame.from_csv('../data/balc_targets.csv', index_col=[0])
pelic = pd.DataFrame.from_csv('../private/pelic_targets.csv', index_col=[0])
targets = pd.DataFrame.from_csv('../exploratory-analysis/chosen_targets_new.csv', index_col=[0])

/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: from_csv is deprecated. Please use read_csv(...) instead. Note that some of the default arguments are different, so please refer to the documentation for from_csv when changing your function calls
  
/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: FutureWarning: from_csv is deprecated. Please use read_csv(...) instead. Note that some of the default arguments are different, so please refer to the documentation for from_csv when changing your function calls
  This is separate from the ipykernel package so we can avoid doing imports until
/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: FutureWarning: from_csv is deprecated. Please use read_csv(...) instead. Note that some of the default arguments are different, so please refer to the documentation for from_csv when changing your function calls
  after removing the cwd from sys.path.


In [3]:
balc.describe()

level_0        index      Filename      Level  token_count  \
count  40.000000    40.000000  4.000000e+01  40.000000    40.000000   
mean   19.500000   927.950000  2.006100e+08   4.500000   211.000000   
std    11.690452   491.064305  6.128939e+03   1.132277    65.332679   
min     0.000000    94.000000  2.006005e+08   3.000000    78.000000   
25%     9.750000   458.000000  2.006060e+08   3.750000   177.250000   
50%    19.500000  1081.000000  2.006081e+08   4.500000   215.000000   
75%    29.250000  1323.750000  2.006116e+08   5.250000   247.750000   
max    39.000000  1563.000000  2.006217e+08   6.000000   397.000000   

             TTR    Guiraud  
count  40.000000  40.000000  
mean    0.500926   7.122289  
std     0.062138   1.131043  
min     0.340081   4.858987  
25%     0.460950   6.178303  
50%     0.491773   7.402602  
75%     0.550922   7.891725  
max     0.608696   9.152086

In [4]:
pelic.describe()

answer_id  question_id  user_file_id   level_id  version  \
count     90.000000    90.000000     90.000000  90.000000     90.0   
mean   20634.766667  2727.966667  48890.066667   4.000000      1.0   
std    12185.777704  1575.875405  22799.671541   0.821071      0.0   
min      539.000000    62.000000   8772.000000   3.000000      1.0   
25%     7865.750000  1203.500000  25620.250000   3.000000      1.0   
50%    23630.000000  3062.000000  56236.000000   4.000000      1.0   
75%    30205.750000  3994.250000  67384.250000   5.000000      1.0   
max    43068.000000  5492.000000  89640.000000   5.000000      1.0   

       toks_re_len    Guiraud  
count    90.000000  90.000000  
mean    213.155556   7.261678  
std     109.302170   1.444528  
min      46.000000   3.502303  
25%     114.250000   6.126377  
50%     191.000000   7.318738  
75%     311.750000   8.393324  
max     400.000000  10.536023

In [5]:
# drop the index column, we don't need it
balc = balc.drop(['index', 'level_0'], axis=1)
balc.head()

Filename  Level                                      Original_Text  \
0  200611403      3  \t\t\t\tCEPA 3 200611403\n\n\n\nLast summer ho...   
1  200608092      4  \t\t\t\tCEPA 4 200608092\n\n\n\nI have just ha...   
2  200603618      5  \t\t\t\tCEPA 5 200603618\n\n\n\nIn the summer ...   
3  200607320      6  \t\t\t\tCEPA 6 200607320\n\n\n\nThere have bee...   
4  200607798      3  \t\t\t\tCEPA 3 200607798\n\nWhere I going to t...   

                                    Normalized_Essay  \
0  Last summer holiday iam going with my family t...   
1  I have just had the perfect holidy. In my holi...   
2  In the summer holiday my family dicides to spe...   
3  There have been many films reteased lately, So...   
4  Where I going to the India with my family , my...   

                                       Revised_Essay  \
0  Last summer holiday iam going with my family t...   
1  I have just had the perfect holidy. In my holi...   
2  In the summer holiday my family dicides to spe...   
3  There have been many films reteased lately, So...   
4  Where I going to the India with my family , my...   

                                              tokens  token_count       TTR  \
0  ['Last', 'summer', 'holiday', 'iam', 'going', ...           82  0.536585   
1  ['I', 'have', 'just', 'had', 'the', 'perfect',...          188  0.446809   
2  ['In', 'the', 'summer', 'holiday', 'my', 'fami...          215  0.460465   
3  ['There', 'have', 'been', 'many', 'films', 're...          234  0.598291   
4  ['Where', 'I', 'going', 'to', 'the', 'India', ...          151  0.602649   

    Guiraud                                                pos  \
0  4.858987  [(Last, 'JJ'), (summer, 'NN'), (holiday, 'NN')...   
1  6.126330  [(I, 'PRP'), (have, 'VBP'), (just, 'RB'), (had...   
2  6.751744  [(In, 'IN'), (the, 'DT'), (summer, 'NN'), (hol...   
3  9.152086  [(There, 'EX'), (have, 'VBP'), (been, 'VBN'), ...   
4  7.405475  [(Where, 'WRB'), (I, 'PRP'), (going, 'VBG'), (...   

                                              lemmas   targ  
0  ['last', 'summer', 'holiday', 'iam', 'go', 'wi...   book  
1  ['-PRON-', 'have', 'just', 'have', 'the', 'per...   book  
2  ['in', 'the', 'summer', 'holiday', '-PRON-', '...   book  
3  ['there', 'have', 'be', 'many', 'film', 'retea...   book  
4  ['where', '-PRON-', 'go', 'to', 'the', 'india'...  money

In [6]:
balc.groupby('Level')['token_count', 'Guiraud'].describe()

token_count                                                         \
            count   mean        std    min     25%    50%     75%    max   
Level                                                                      
3            10.0  151.7  64.187313   78.0  102.25  144.0  167.00  264.0   
4            10.0  211.3  41.411351  138.0  185.75  209.0  238.00  286.0   
5            10.0  251.6  70.062195  172.0  188.75  255.0  282.25  397.0   
6            10.0  229.4  40.467271  183.0  200.75  220.5  243.00  303.0   

      Guiraud                                                              \
        count      mean       std       min       25%       50%       75%   
Level                                                                       
3        10.0  6.163016  1.109401  4.858987  5.306816  5.996125  7.194436   
4        10.0  6.811704  1.021188  5.344792  6.127009  6.680361  7.663461   
5        10.0  7.484633  0.928882  6.186455  6.807033  7.507225  7.826307   
6        10.0  8.029801  0.465230  7.529412  7.798222  7.981108  8.169718   

                 
            max  
Level            
3      7.877855  
4      8.455767  
5      8.944468  
6      9.152086

In [7]:
pelic.groupby(['level_id', 'native_language'])['toks_re_len', 'Guiraud'].describe()

toks_re_len                                           \
                               count   mean         std    min     25%    50%   
level_id native_language                                                        
3        Arabic                 10.0  191.4  105.506398   75.0  113.50  174.0   
         Korean                 10.0  187.0   79.033045  100.0  123.25  171.0   
         Spanish                10.0  152.3   80.963709   69.0  102.00  139.5   
4        Arabic                 10.0  258.9  121.135782   57.0  165.50  303.0   
         Korean                 10.0  218.1  107.858395   46.0  170.25  219.5   
         Spanish                10.0  187.1  137.795702   59.0   86.25  106.0   
5        Arabic                 10.0  226.0  103.297843   79.0  131.00  246.0   
         Korean                 10.0  180.6  107.968308   60.0   96.50  159.0   
         Spanish                10.0  317.0   68.821831  159.0  303.00  330.0   

                                        Guiraud                                \
                             75%    max   count      mean       std       min   
level_id native_language                                                        
3        Arabic           234.00  400.0    10.0  6.504896  1.098519  4.638007   
         Korean           239.50  330.0    10.0  7.221210  1.824843  3.502303   
         Spanish          169.75  355.0    10.0  6.011871  0.666962  5.151778   
4        Arabic           359.50  392.0    10.0  7.592946  1.402359  4.900770   
         Korean           309.25  361.0    10.0  7.587213  1.240380  5.602794   
         Spanish          312.25  398.0    10.0  7.127076  1.926144  5.077368   
5        Arabic           297.50  366.0    10.0  7.722298  1.377498  5.659800   
         Korean           215.50  374.0    10.0  7.333414  1.370396  4.776679   
         Spanish          361.50  397.0    10.0  8.254176  0.832434  6.661633   

                                                                   
                               25%       50%       75%        max  
level_id native_language                                           
3        Arabic           6.181381  6.649604  7.135215   7.848084  
         Korean           6.070947  7.876158  8.275595   9.358192  
         Spanish          5.589022  5.930714  6.427172   7.295815  
4        Arabic           6.665213  8.151374  8.391443   9.564674  
         Korean           6.758256  7.602113  8.331711   9.353040  
         Spanish          5.648795  6.457281  8.270508  10.536023  
5        Arabic           6.553237  7.899653  8.740135   9.616783  
         Korean           6.857697  7.363757  8.053800   9.499716  
         Spanish          7.744835  8.385115  8.674723   9.712147

In [8]:
def getArticles(regex, text):
    """Uses a regex targeting a noun. Finds all instances of the noun and returns whether the return contains a definite article, an indefinite article, a zero article, or other types of determiners.
    Input: regex (a regex compile), text (can be a single string object or a list of string objects).
    Returns item and type of article in a list format."""
    myre = re.compile(regex)
    look = myre.findall(text)
    other = ['my', 'his', 'her', 'their', 'their', 'our', 'ours', 'its', 
            'this', 'that', 'these', 'those', 'some', 'much', 'many', 'most', 'some', 'any', 'enough',
            'all', "both", 'half', 'either', 'neither', 'each', 'every']
    articles = []
    defi = re.compile(r'\bthe\b')
    indef = re.compile(r'\ban?\b')
    for item in look:
        if any(search in item[0] for search in other):
            if defi.search(item[0]):
                articles.append((item[0], 'the'))
            elif indef.search(item[0]):
                matchobj = indef.search(item[0])
                if matchobj:
                    articles.append((item[0], 'a/an'))
                else:
                    articles.append((item[0], 'other'))
            else:
                articles.append((item[0], 'other'))
        elif defi.search(item[0]):
            articles.append((item[0], 'the'))
        elif indef.search(item[0]):
            matchobj = indef.search(item[0])
            if matchobj:
                articles.append((item[0], 'a/an'))
        else:
            articles.append((item[0], '∅'))
    return articles

In [9]:
test = ["I put the book on the counter.", "I put a book on the counter.", "I put the books on the counter",
       "I put some books on the counter.", "I put my books on the counter.", "I put book on the counter.",
       "I put books on the counter.", "I put book and books and the other books on the counter."]
myre = re.compile(r'((\w+ ){,2}\w+ book.*?\b)', re.I)

for s in test:
    print(getArticles(myre, s))

[('I put the book', 'the')]
[('I put a book', 'a/an')]
[('I put the books', 'the')]
[('I put some books', 'other')]
[('I put my books', 'other')]
[('I put book', '∅')]
[('I put books', '∅')]
[('I put book', '∅'), ('and books', '∅'), ('and the other books', 'the')]


## Analysis

### PELIC
First, I'll break things up 

In [10]:
pelic3 = pelic[pelic.level_id == 3]
pelic4 = pelic[pelic.level_id == 4]
pelic5 = pelic[pelic.level_id == 5]

#### Level 3

In [11]:
pelic3 = pelic3.reset_index()
pelic3 = pelic3.drop(27, axis=0)

In [12]:
# for x in range(len(pelic3)):
#     print(pelic3.answer_id[x], ":", pelic3.native_language[x])
#     print(pelic3.text[x])
#     print()
#     print("----------")
#     print()

In [14]:
pelic3[pelic3.native_language == 'Arabic'].sample(5, random_state=3)

index  answer_id  question_id anon_id  user_file_id  \
18     54      26339         3452     et5         61494   
6      18      35163         4575     cr8         75072   
21     63      20934         2789     dw0         47858   
12     36      26402         3451     da7         61537   
15     45      27626         3672     cj7         63318   

                                                 text class_code  level_id  \
18        Two heads are better than one\nWorking a...          w         3   
6   \nParents play the most important role in the ...          w         3   
21     The money is the root of all evil. First, n...          w         3   
12          Haste makes waste. I really agree with...          w         3   
15  1.        Lion  live in jungle and deer does  ...          w         3   

   native_language  version  \
18          Arabic        1   
6           Arabic        1   
21          Arabic        1   
12          Arabic        1   
15          Arabic        1   

                                              toks_re  toks_re_len  \
18  ['two', 'heads', 'are', 'better', 'than', 'one...           75   
6   ['parents', 'play', 'the', 'most', 'important'...          400   
21  ['the', 'money', 'is', 'the', 'root', 'of', 'a...           90   
12  ['haste', 'makes', 'waste', 'i', 'really', 'ag...          137   
15  ['lion', 'live', 'in', 'jungle', 'and', 'deer'...          113   

                                            toks_nltk   Guiraud   targ  
18  ['Two', 'heads', 'are', 'better', 'than', 'one...  6.119913   week  
6   ['Parents', 'play', 'the', 'most', 'important'...  7.800000    sea  
21  ['The', 'money', 'is', 'the', 'root', 'of', 'a...  4.638007  money  
12  ['Haste', 'makes', 'waste', '.', 'I', 'really'...  6.834861   time  
15  ['1', '.', 'Lion', 'live', 'in', 'jungle', 'an...  4.797676  water

#### Format of analysis: 
- word/phrase: (produced, expected), rule

Nominal objects that are modified by other determiners (such as possessives) will not be analyzed for this project. The rules and cues discussed in this portion of the analysis are derived from Zhao and MacWhinney (2018) and Cowan (2008). 

To begin, only a small sample of data will be analyzed, given the large nature of the starting point. To begin, I will work with five from each level.

I have opted to drop certain columns from the data for privacy or complete lack of understanding.

In [38]:
pelic3.text[18]

"      Two heads are better than one\nWorking as a group is always better than an individual. Last week, my friends and I went to a soup kitchen to serve food to people who can't get food easily. First, we prepared the place for them. Next, we ushered them to their tables. Then, we served them the food. Finally, we cleaned the place. So we didn't feel tired because we worked as a group.   \n"

Subject: 34717 (Arabic)
- working: (∅, ∅), general activity
- group: (a, a), ?
- individual: (an, an), ?
- soup kitchen: (a, a), first mention
- food: (∅, ∅), noncount
- people:  (∅, ∅), noncount
- food: (∅, ∅), noncount 
- place: (the, the), second mention (with variation)
- food: (the, ?), second mention specific
- place: (the, the), second mention (with variation)
- group: (a, a), collective

In [18]:
pelic3.text[6]

"\nParents play the most important role in the growing up of their children. In my opion, despite some very good characteristics parents have, they can't be the best teachers of their children.\n\nFirst of all, parents are able to have knowledge in some fields, but not all the fields. It is not true that parents know everything in the world. Sometimes, they just have a passing acquaintance of some subjects. They shouldn't teach their children something that they don't know thoroughly. For example, a very famous physicist can't teach his children chemistry though he is very good at physics because his knowledge of chemistry isn't enough to teach his children. So that's why parents always search for good teachers to teach their children what they need to learn rightly.\n\nSecondly, parents want to teach their children by their ways. They are right, and they are wrong sometimes. For those who aren't be educated how to be good teachers, it might be a very big problem. They can be good pare

35163 : Arabic

- parents: (∅, ∅), generic reference
- role: (the, the), superlative
- growing up: 
- characteristics: 
- teachers: 
- parents: 
- knowledge: 
- fields: 
- fields: 
- parents: 
- world: 
- acquaintance: 
- subjects: 
- physicist: 
- chemistry: 
- physics: 
- chemistry: 
- teachers: 
- parents: 
- teachers: 
- problem: 
- parents: 
- teachers: 
- children: 
- children: 
- explanation: 
- parents: 
- wrong: 
- right: 
- parents: 
- educating: 
- best ones: 
- things: 
- children: 
- parents: 
- situation: 
- children: 
- parents: 
- right: 
- parents: 
- teachers: 
- parents: 
- teachers: 
- way: 
- way: 

In [15]:
pelic3.text[12]

'        Haste makes waste. I really agree with this statement. I had a lot of personal experience in this. For example, I studied at a university. One day, my teacher wanted to give the final exam for our class. I was surprised about that because he did not give us any lessons throughout the term. I asked him how I could take his exam if I did not learn any thing in his class. After that, he appeared very nervous. I do not know why. Next, I went to his office, and he began to curse at me. I know his reaction was very bad, but this was his mistake and I could not do any thing to pass this course. In the end, I know now I was not having experience in this situation at that time. '

In [34]:
pelic3.text[21]

'   The money is the root of all evil. First, now the money became very evil in all the people. Second, the money is loss the friend. Third, the money is attracting wrestling between the people. Then, the money is very important each other people. Next, the people are ambition for the money. Finally, the money turns up enmity between the people. The money is important for everyone, but some of the people are going to jell because the money. Also all people working very hard every day for the money'

20934 (Arabic)
- money: (the, ∅), noncount / generic reference
- money: (the, ∅), noncount / generic reference
- people: (the, the), marked but not ungrammatical (imo)
- money: (the, ∅), noncount / generic reference
- friend: (the, ∅), should be plural
- people: (the, ∅), noncount / generic reference
- wrestling: (∅, ∅), noncount
- people: (the, ∅), noncount / generic reference
- money: (the, ∅), noncount / generic reference
- people: (the, ∅), noncount / generic reference
- ambition: (∅, ∅), noncount
- money: (the, ∅), noncount / generic reference
- money: (the, ∅), noncount / generic reference
- enmity: (∅, ∅), noncount
- people: (the, ∅), noncount / generic reference
- money: (the, ∅), noncount / generic reference
- people: (the, ∅), noncount / generic reference
- jell (jail): (∅, ∅), noncount
- money: (the, ∅), noncount / generic reference
- money: (the, ∅), noncount / generic reference

In [40]:
pelic3.text[15]

"1.        Lion  live in jungle and deer does  too\n2.        Lion have fur and so does deer.                         \n3.         Deer live with group and lion does too.\n4.         Both lion and deer drink water.\n5.        The lion has same color as deer\n6.        Deer feet the baby like lion        \n7.        Lion is as fast speed as deer\n8.        Similar too the Loin, deer  sleep at night  \n\n1.        Loin bigger than deer \n2.        The lion eat meat and deer doesn't  too\n3.        the loin  is stronger  than the deer \n4.        the deer cuter than lion\n5.        loin killer and deer doesn't too\n6.        deer eat grass but the loin eat deer\n7.        the deer taller on the other hand the loin shorter  \n8.        loin has life  different then deer \n"

27626 (Arabic) 
- lion: (∅, the) -- should be lion**s** with ∅
- jungle: (∅, the), 
- deer: (∅, a) -- but they think it's singular ('does')
- lion: (∅, the) -- should be lion**s** with ∅
- fur: (∅, ∅)
- deer: (∅, a) -- but they think it's singular ('does')
- deer: (∅, ∅)
- group: (∅, a) -- needs to be **plural** with ∅
- lion: (∅, the) -- should be lion**s** with ∅
- lion: (∅, ∅) -- should be lion**s**
- deer: (∅, ∅) 
- water: (∅, ∅) 
- lion: (the, the) 
- deer: ?
- deer: (∅, ∅) 'feed'
- baby: (the, a) -- needs to be **plural** with ∅
- lion: (∅, the) -- should be lion**s** with ∅
- lion: (∅, the) -- should be lion**s** with ∅
- speed: ?
- deer: (∅, ∅) 
- loin (lion): (the, the) 
- deer: (∅, ∅) 
- night: (∅, ∅) 
- loin (lion): (∅, the) -- should be lion**s** with ∅
- deer: (∅, ∅) ?
- lion: 
- meat: 
- deer: 
- loin: (the, the)
- deer: (the, the) 
- deer: (the, the)
- lion: (∅, the) -- should be lion**s** with ∅
- lion
- killer 
- deer 
- deer
- grass
- lion
- deer 
- deer
- lion
- lion 
- life
- deer 

In [15]:
pelic3[pelic3.native_language == 'Spanish'].sample(5, random_state=1)

index  answer_id  question_id anon_id  user_file_id  \
8      24      30327         4007     ci4         67477   
29     87      29438         3921     ci4         66641   
20     60       6229          975     cw2         23603   
14     42       6959         1020     cw2         24828   
2       6      29216         3881     bt8         66247   

                                                 text class_code  level_id  \
8                              My ideal Apartament...          w         3   
29  \nTravel\n\nThe people are travel around the w...          w         3   
20  About my favorite restaurant.\n\n      My favo...          w         3   
14  MY PROUDEST MOMENT\n\n       My proudest momen...          w         3   
2                                                 ...          w         3   

   native_language  version  \
8          Spanish        1   
29         Spanish        1   
20         Spanish        1   
14         Spanish        1   
2          Spanish        1   

                                              toks_re  toks_re_len  \
8   ['my', 'ideal', 'apartament', 'my', 'ideal', '...          171   
29  ['travel', 'the', 'people', 'are', 'travel', '...          184   
20  ['about', 'my', 'favorite', 'restaurant', 'my'...           69   
14  ['my', 'proudest', 'moment', 'my', 'proudest',...          114   
2   ['types', 'of', 'magazines', 'there', 'are', '...          166   

                                            toks_nltk   Guiraud  targ  
8   ['My', 'ideal', 'Apartament', 'My', 'ideal', '...  5.964809   sea  
29  ['Travel', 'The', 'people', 'are', 'travel', '...  5.160468  park  
20  ['About', 'my', 'favorite', 'restaurant', '.',...  5.778521  week  
14  ['MY', 'PROUDEST', 'MOMENT', 'My', 'proudest',...  5.525856  time  
2   ['Types', 'of', 'magazines', 'There', 'are', '...  7.295815  book

In [14]:
pelic3[pelic3.native_language == 'Korean'].sample(5, random_state=1)

index  answer_id  question_id anon_id  user_file_id  \
7      21      26815         3495     cn3         62307   
28     84      26116         3422     cn3         60702   
19     57      13572         1931     fj7         33057   
13     39      27628         3673     cn3         63320   
1       3      18572         2544     dg7         41011   

                                                 text class_code  level_id  \
7   Foreign Student Majors\n\n I want to open an e...          w         3   
28   What makes me decide which shoes to buy?\nWhe...          w         3   
19   American Billboard Chart have many kinds musi...          w         3   
13  Superman and Batman have several things in com...          w         3   
1   ?        LIST\n\nThe entrance is in front of t...          w         3   

   native_language  version  \
7           Korean        1   
28          Korean        1   
19          Korean        1   
13          Korean        1   
1           Korean        1   

                                              toks_re  toks_re_len  \
7   ['foreign', 'student', 'majors', 'i', 'want', ...          107   
28  ['what', 'makes', 'me', 'decide', 'which', 'sh...          123   
19  ['american', 'billboard', 'chart', 'have', 'ma...          100   
13  ['superman', 'and', 'batman', 'have', 'several...          177   
1   ['list', 'the', 'entrance', 'is', 'in', 'front...          124   

                                            toks_nltk   Guiraud  targ  
7   ['Foreign', 'Student', 'Majors', 'I', 'want', ...  6.283787   sea  
28  ['What', 'makes', 'me', 'decide', 'which', 'sh...  5.680519  park  
19  ['American', 'Billboard', 'Chart', 'have', 'ma...  6.000000  week  
13  ['Superman', 'and', 'Batman', 'have', 'several...  7.741954  time  
1   ['?', 'LIST', 'The', 'entrance', 'is', 'in', '...  3.502303  book

18572 : Korean

I have little furniture in my room.
As you enter the door through the entrance, desk is in front of you. The desk is on the front, between book chests. I have two book chests. One book chest is on the right wall. The other one is on the front wall. My bed is on the left corner, beside book chest. Clothes chest is on the left, in front of bed. My room center has the heart shape rug.

- furniture: (∅, ∅), 
- room: my 
- door: (the, the), general knowledge (??)
- entrance: (the, the), there's only one
- desk: (∅, a), first mention
- desk: (the, the), second mention
- front: (the, the), front of the room
- book chests: (∅, ∅), plural
- book chests: two 
- book chest: one
- wall: (the, the), one of four 
- other one: other
- wall: (the, the), one of four 
- bed: my 
- left corner: (the, the), one of four 
- book chest: (∅, the), second mention
- clothes chest: (∅, a), first mention
- left: (the, the)
- bed: (∅, the), second mention 
- room center: my
- heart shape rug: (the, a), first mention

29216 : Spanish

There are three types of magazines, they are available from all good bookshops or online.  I can to get the best preference or the information about if it is interesting for me.

The first type of magazine is a sport magazine.  For example, I can find magazine providing the latest information on building and operating an athletic facility, athletic programs, action photos, athlete interviews, scores and statistic on an  professional and college sports.  The second type of magazine is an automobile whether if I have a passion for cars, I can find a long list of magazines especially tailored for the information needs of car such as car design, car market, or car lifestyle.  The last type is a health magazine these cover a variety of topics including physical fitness, nutrition, beauty, weight training and body building.
In conclusion, I can find many types of magazines the most important,  I can choose or find my preference and what information is interesting for me.

14646 : Korean
        
I have a lovely sister who works at a bank in Korea. She is two years younger than me, but she is as tall as me. Of course, she is lighter than I am in weight. Then, she looks younger than her real age; I think it is because of the history of our family. Actually, all of my family members, including me, have young faces. Next, she has splendid, long black hair. Also, she often likes to change her hair style. So, she sometimes has big wavy long hair, or sometimes she has straight long hair. In addition, she has big eyes and double-edged eyelids, unlike me. Moreover, she has bright skin color and a good smile. Therefore, people who have ever seen her think that she is feminine and benevolent.
However, I know that she is not as affectionate as her appearance. First of all, she is selfish and doesn't take a conciliatory attitude to me. Especially, she yields an inch on that food, so we sometimes squabble. Once, someone came to my home with Haagen Dazs Vanila ice cream, and we put it in our refrigerator. At that time, I had no time to eat it because I had to go out to finish something. I was only thinking of the ice cream while I went out. However, I didn't eat it because my sister ate it all. I was really angry and asked for her about it. Then, she said, "Why didn't you eat it before I eat it?". On the contrary, she sometimes makes me laugh because she mishears what I say. Once, I asked her, "Do you try the cheese snack?", but she misheard me. She heard "employment" instead of "cheese snack", "employment" is pronounced as 'chui jik' in Korean. Also, Such things are a daily event. Finally, I wish she would conduct in her appearance. Then, I think that she can be more lovely woman than what she is. 

19745 : Spanish

Both "Mc Donald's" and "Wendy's" are restaurants that sell fast food. First of all, Mc Donald's is as delicious as Wendy's. Next, Wendy's is famous around the U.S. and so is Mc Donald's. Both restaurants have similar food and both restaurants are very common. Then both "Mc Donald's" and "Wendy's" prepare very special hamburgers. Finally both restaurants are very nice. Come and try their hamburgers, then you can choose which one is special for you and know why I said that. The two restaurants serve a great fast food.  


26815 : Korean
Foreign Student Majors

 I want to open an english langage school. I researched about foreign students' majors. They have many kinds of majors. The most popular field is business and management. It is 19.4%. The second is engineering, and it's 15.2%. 
In addition, mathematics and computer sciences, social sciences and physical and life sciences are 12%, 7.7% and 7%, respectively. These three fields are kinds of sciences. So, the sum of the three sciences is 26.7%. It is a bigger than business and management or engineering. According to this result, I need to open special classes for technical english. Many students want to study english that can be  used in their careers. 


30327 : Spanish
                           My ideal Apartament 

My ideal apartament will be a perfect and confortable place for me.
this apartament will everything want. first, my ideal apartament is a big apartament in Miami Beach, with five rooms. The principal room be for me with a big closet and big bathroom with jacuzzi. then, other 2 rooms for my friends and family when they going to visit me in my apartament. Other room is the studio room with all my works things. the last one is the service room. Second, my ideal apartament go to smell so good always, and the view in my apartament will be the sea. Third, the decoration in my apartament will be the colors red black and white, with a modern style. When i will stay in my apartament i go to feel like to a queen. finally, every day i will eat my favorite food because in my apartament the chef will cook. My ideal apartament will be perfect and confortable place for me, with all things i describe before. 
 